test/transcoding/IntelFPGAMemoryAccesses.ll

```diff
@@ -0,0 +1,218 @@
+; LLVM IR generated by Intel SYCL Clang compiler (https://github.com/intel/llvm)
+; SYCL source code can be found below:
+
+; struct State {
+;  float Field1;
+;  int Field2;
+;  };
+;
+; void foo(float *A, int *B, State *C) {
+;   float *x;
+;   int *y;
+;   State *z;
+;   x = __builtin_intel_fpga_mem(A, 1 | 2, 0);
+;   y = __builtin_intel_fpga_mem(B, 4 | 8, 0);
+;   z = __builtin_intel_fpga_mem(C, 2, 127);
+;   x = __builtin_intel_fpga_mem(&C->Field1, 1 | 2, 127);
+;   y = __builtin_intel_fpga_mem(&C->Field2, 0, 127);
+;   z = __builtin_intel_fpga_mem(C, 1 | 2 | 4 | 8, 127);
+; }
+;
+; template <typename name, typename Func>
+; __attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+;   kernelFunc();
+; }
+;
+; int main() {
+;   kernel_single_task<class fake_kernel>([]() {
+;     float *A;
+;     int *B;
+;     State *C;
+;     foo(A, B, C);
+;   });
+;   return 0;
+; }
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_fpga_memory_accesses -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Capability FPGAMemoryAccessesINTEL
+; CHECK-SPIRV: Extension "SPV_INTEL_fpga_memory_accesses"
+; CHECK-SPIRV: Decorate {{[0-9]+}} BurstCoalesceINTEL
+; CHECK-SPIRV: Decorate {{[0-9]+}} DontStaticallyCoalesceINTEL
+; CHECK-SPIRV: Decorate {{[0-9]+}} PrefetchINTEL 0
+; CHECK-SPIRV: Decorate {{[0-9]+}} CacheSizeINTEL 0
+; CHECK-SPIRV: Decorate {{[0-9]+}} CacheSizeINTEL 127
+; Check that the semantically meaningless decoration was
+; translated as a mere annotation
+; CHECK-SPIRV: MemberDecorate {{[0-9]+}} 1 UserSemantic "{params:0}{cache-size:127}"
+; Check regular struct member decorations
+; CHECK-SPIRV: MemberDecorate {{[0-9]+}} 0 BurstCoalesceINTEL
+; CHECK-SPIRV: MemberDecorate {{[0-9]+}} 0 CacheSizeINTEL 127
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown-sycldevice"
+
+%"class._ZTSZ4mainE3$_0.anon" = type { i8 }
+%struct._ZTS5State.State = type { float, i32 }
+
+; CHECK-LLVM: [[PARAM_3_CACHE_0:@\.str[0-9_.]*]] = {{.*}}{params:3}{cache-size:0}
+@.str = private unnamed_addr constant [25 x i8] c"{params:3}{cache-size:0}\00", section "llvm.metadata"
+@.str.1 = private unnamed_addr constant [14 x i8] c"<invalid loc>\00", section "llvm.metadata"
+; "params" bitmask doesn't hold 0x2 (cache size ON), so cache-size can be dropped
```

Comment:
I think this logic shall be mirrored to the translator sources.

Reply:
It actually is - `{cache-size:..}` isn't generated when the `CacheSizeINTEL` decoration is not discovered in SPIR-V.

Reply:
Oh, I meant to be mirrored with the appropriate comment.

Reply:
@MrSidims, could you please take a look at the updated patch and make a GH suggestion if your comment persists? I have to confess I couldn't figure out an appropriate resolution here.